src/services/db/GitFileCommitService.ts

```diff
@@ -36,51 +56,43 @@ export default class GitFileCommitService {
       isMedia?: boolean
     }
   ): Promise<{ sha: string }> {
-    const createPromises = [
-      this.gitFileSystemService.create(
+    const stagingCreateResult = await this.gitFileSystemService.create(
+      sessionData.siteName,
+      sessionData.isomerUserId,
+      content,
+      directoryName,
+      fileName,
+      isMedia ? "base64" : "utf-8",
+      STAGING_BRANCH
+    )
+    const shouldUpdateStagingLite =
+      isReduceBuildTimesWhitelistedRepo(sessionData.growthbook) &&
+      !isFileAsset({ directoryName, fileName })
+
+    let stagingLiteCreateResult
+    if (shouldUpdateStagingLite) {
+      stagingLiteCreateResult = await this.gitFileSystemService.create(
         sessionData.siteName,
         sessionData.isomerUserId,
         content,
         directoryName,
         fileName,
         isMedia ? "base64" : "utf-8",
-        STAGING_BRANCH
-      ),
-    ]
-    const shouldUpdateStagingLite =
-      isReduceBuildTimesWhitelistedRepo(sessionData.growthbook) &&
-      !isFileAsset({ directoryName, fileName })
-
-    if (shouldUpdateStagingLite) {
-      createPromises.push(
-        this.gitFileSystemService.create(
-          sessionData.siteName,
-          sessionData.isomerUserId,
-          content,
-          directoryName,
-          fileName,
-          isMedia ? "base64" : "utf-8",
-          this.STAGING_LITE_BRANCH
-        )
+        STAGING_LITE_BRANCH
       )
     }
-    const [stagingCreateResult, stagingLiteCreateResult] = await Promise.all(
-      createPromises
-    )
 
     if (stagingCreateResult.isErr()) {
       throw stagingCreateResult.error
-    } else if (shouldUpdateStagingLite && stagingLiteCreateResult.isErr()) {
+    } else if (
+      shouldUpdateStagingLite &&
+      stagingLiteCreateResult &&
+      stagingLiteCreateResult.isErr()
+    ) {
       throw stagingLiteCreateResult.error
     }
 
-    this.gitFileSystemService.push(sessionData.siteName, STAGING_BRANCH)
-    if (shouldUpdateStagingLite) {
-      this.gitFileSystemService.push(
-        sessionData.siteName,
-        this.STAGING_LITE_BRANCH
-      )
-    }
+    this.pushToGithub(sessionData, shouldUpdateStagingLite)
```

Comment:
this can fail silently - we should probably just chain off this. alternatively, what we can do is just chain off the results sequentially and throw an error at the end (the behaviour now), which makes things clearer to read. 

also eliminates the issue of `staging-lite` updating but not `staging` cos if the initial call to `staging` fails, `staging-lite` won't even run

Reply:
Hey this was an intentional trade off we made as a team not to await for calls for they are expensive. But I do air your concerns regarding silent failures, this is the cause of the divergence that we are seeing. Dont intend to tackle it here!

For info only + outside of scope of pr: I tried adding an alarm for this, but it was TOO noisy, there is prob a way to make this such that we only throw alarms for this silent pushes, but havnt had time to invetigate

Reply:
we have a few different ways to tackle this tbh, just spitballing ideas - not an ask to implement. 

1. separate out the push to github and let the router handle this. the router can always send response early then fail later (no cost to user) if the push to github fails. 
2. because `neverthrow` is safe, we could just write a wrapper for retrying `Errors<T>`
3. just send an email on failure 

we probably want to decide what categories of errors we care about - transient errors vs persistent errors; in terms of our usage, transient is probably outages <10 mins in length (we don't have that much pushes tbvh) and persistent >=10. 

if transient, retry w/ exponential backoff will work well; if persistent, we probably want a notif + manual fix. to be honest, the divergence should arise only if we edit on github directly whilst there's a push failure so should be minimal but the problem is we got high barrier of entry to edit on efs (hence why we edit on github directly).